modules/utxo-lib/src/inscriptions.ts

```diff
@@ -0,0 +1,66 @@
+import * as assert from 'assert';
+import { Payment, script as bscript } from 'bitcoinjs-lib';
+import { p2tr } from './payments';
+import { ecc as eccLib } from './';
+
+const OPS = bscript.OPS;
+const MAX_LENGTH_TAP_DATA_PUSH = 520;
+
+/**
+ * The max size of an individual OP_PUSH in a Taproot script is 520 bytes. This
+ * function splits inscriptionData into an array buffer of 520 bytes length.
+ * https://docs.ordinals.com/inscriptions.html
+ * @param inscriptionData
+ */
+function sliceInscriptionData(inscriptionData: Uint8Array) {
+  if (inscriptionData.length <= MAX_LENGTH_TAP_DATA_PUSH) {
+    return [Buffer.from(inscriptionData)];
+  }
+
+  const pushDataBuffers: Buffer[] = [];
+  for (let i = 0; i < inscriptionData.length; i += MAX_LENGTH_TAP_DATA_PUSH) {
+    const fixedLengthChunk = inscriptionData.slice(i, i + MAX_LENGTH_TAP_DATA_PUSH);
+    pushDataBuffers.push(Buffer.from(fixedLengthChunk));
+  }
+
+  return pushDataBuffers;
+}
+
+/**
+ * Generate an inscription address.
```

Comment:
To stay more consistent with our other address generation methods (`src/bitgo/outputScripts.ts`), let's make the result an output script (`Buffer`) here as well.

One of the advantages is that output scripts are not coin-dependent so the caller can convert to the desired address format. This code here assumes bitcoin (mainnet?).

Reply:
Also we can already think about spending from the output. We can extract a method `createRedeemScriptForInscription(...): Buffer` from here.